pkg/hubble/parser/threefour/parser.go

```diff
@@ -435,6 +432,29 @@ func decodeCiliumEventType(eventType, eventSubType uint8) *pb.CiliumEventType {
 	}
 }
 
+func decodeTraceReason(tn *monitor.TraceNotify) pb.TraceReason {
+	if tn == nil {
+		return pb.TraceReason_TRACE_REASON_UNKNOWN
+	}
+	// The protobuf enum values aren't 1:1 mapped with Cilium's datapath
+	// because we want pb.TraceReason_TRACE_REASON_UNKNOWN = 0 while in
+	// datapath monitor.TraceReasonUnknown = 5. The mapping works as follow:
+	switch {
+	// monitor.TraceReasonUnknown is mapped to pb.TraceReason_TRACE_REASON_UNKNOWN
+	case tn.TraceReason() == monitor.TraceReasonUnknown:
+		return pb.TraceReason_TRACE_REASON_UNKNOWN
+	// values before monitor.TraceReasonUnknown are "offset by one", e.g.
+	// TraceReasonCtEstablished = 1 → TraceReason_ESTABLISHED = 2 to make room
+	// the the zero value.
+	case tn.TraceReason() < monitor.TraceReasonUnknown:
+		return pb.TraceReason(tn.TraceReason()) + 1
+		// all values greater than monitor.TraceReasonUnknown are mapped 1:1
+		// with the datapath values.
+	default:
+		return pb.TraceReason(tn.TraceReason())
+	}
```

Comment:
woof. Makes sense after I actually checked out the code and looked at values, sorry you had to deal with this.